### PR TITLE
Selected ItemStack: Reduce black magic

### LINKED ISSUE #6516: Strange behaviour when manipulating full stacks in chest formspecs
Tested in Minetest and Minetest_game 0.4.16.

Put down a chest. Get hold of two stacks of 99 cobblestone each, plus one stack of 1 cobblestone. Put one stack of 99 cobblestone in the chest, the second in your first hotbar slot, and the stack of 1 cobblestone in your second hotbar slot. Now open the chest, click on the 1-cobble stack from hotbar slot 2 to pick it up into the mouse pointer, and then click on the 99-cobble stack in the chest.

Expected behaviour: the stacks swap places. The chest contains a 1-cobble stack, the mouse pointer holds a 99-cobble stack, and the 99-cobble stack in hotbar slot 1 is still there.

Actual behaviour: The chest contains a 1-cobble stack, the 99-cobble stack from the chest teleports directly to hotbar slot 2, and the 99-cobble stack from hotbar slot 1 teleports into the mouse pointer.

If you had a >1-cobble stack in hotbar slot 2 when you started this, and you split that stack to put only some of it into the chest, then the stack unsplits when you click in the chest’s inventory and the left-behind portion teleports into the chest at the same time as the 99-cobble stack you swapped with teleports back to hotbar slot 2.

This doesn’t require that the other 99-stack be in the player inventory; it also happens if the 99-stack that you’re not touching is in the chest. However, it *does* require that the 99-stack that you’re not touching be in an earlier slot than the 1-stack that you initially pick up; if there is only one 99-stack before the 1-stack (the one that you’re going to swap with), then the swap happens normally and 99-stacks in higher slots than the original 1-stack lived in are not touched.

### PR DESCRIPTION
**Pull request by SmallJoker**
Better button variable value naming
Code style cleanup. Actual changes happen in `updateSelectedItem()` and in `OnEvent()` (L3500 and further)

Fixes #6516 

Thanks to (nested?) async calls to the player inventory in OnEvent, we still need some black magic to skip the next item verification function to ensure the new item is read. Another option is a recursive mutex, which breaks constant functions in the inventory. After hours of trying to fix this issue using locks without success, I believe it's best to reduce the magic for now.

**Instructions:**
Test this with an pre-PR build first, then post-PR.
1) 2x entire stack of item1 in the chest (horizontally LTR, for these instructions)
2) 2x entire stack of item2 in your inventory (horizontally LTR, for these instructions)
3) Select 2nd stack from inventory (item2), replace with 2nd stack from chest (item1)
4) Restore old setup (1 and 2) and do 3) again
5) Pre-PR behaviour:
   * Chest: [item1] [item2]
   * Inventory: [item2] [item1]
   * Selected: [item1] (1st from chest)
6) Post-PR behaviour:
   * Chest: [item1] [item2] (unchanged)
   * Inventory: [item2] [item1] (unchanged)
   * Selected: [item1] (2nd from inventory)

### CODE REVIEW

File: src/guiFormSpecMenu.cpp [hunk 1]
```diff
@@ -3580,6 +3536,20 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 			a->to_list = s.listname;
 			a->to_i = s.i;
 			m_invmgr->inventoryAction(a);
+
+			if (swap_items) {
+				if (button == MB_RIGHT)
```
Comment (SmallJoker) [inline src/guiFormSpecMenu.cpp]: Note to reviewers: A little feature injection to set the new selected stack size depending on which button is pressed.

**Comment by Hawk777** [comment]
It’s still possible to get some funny behaviour when rapidly swapping itemstacks between mouse pointer and chest grid (perhaps due to the network lag originally theorized in #6516). However, with this commit, the original problem described in #6516 is solved. Thanks!

**Comment by NewbProgrammer101** [comment]
Merge Merge Merge Merge MergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMergeMerge

**Comment by asl97** [comment]
@SmallJoker There is the funny behaviour that @Hawk777 mention,

If I understood "breaks constant functions" correctly, the answer isn't some kind of lock.

It's a flag to prevent it from messing with the selected amount until it been probably swapped. something like: https://github.com/asl97/minetest/commit/35e3d3576247ce1dd0b305deba481e281f66f913

**Comment by SmallJoker** [comment]
Rebased, added testing instructions.

**Pull request merged**